spec/models/miq_report/generator_spec.rb

```diff
@@ -0,0 +1,125 @@
+                                                                     :hours => (1...24)})
+    @host1 = FactoryGirl.create(:host, :name => "Test Host 1")
+    @host2 = FactoryGirl.create(:host, :name => "Test Host 2")
+  end
+  
+  describe "#generate " do
+    context "Memory Utilization Trends report (daily)" do
+      before do
+        @miq_report_profile_all = MiqReport.new(
+          :name       => "Host Memory Utilization Trends (last week)",
+          :title      => "TEST Custom VM report",
+          :rpt_group  => "Custom",
+          :rpt_type   => "Custom",
+          :db         => "VimPerformanceTrend",
+          :order      => "Ascending",
+          :sortby     => ["resource_name"],
+          :time_profile_id => @time_profile_all.id,
+          :db_options => {:limit_col    => "max_derived_memory_available",
+                          :trend_col    => "derived_memory_used",
+                          :rpt_type     => "trend",
+                          :start_offset => 604800,
+                          :end_offset   => 0,
+                          :trend_db     => "HostPerformance",
+                          :interval     => "daily",
+                          :target_pcts  => [70, 80, 90]})
+        
+        @miq_report_profile_exl = MiqReport.new(
+            :name       => "Host Memory Utilization Trends (last week)",
+            :title      => "TEST Custom VM report",
+            :rpt_group  => "Custom",
+            :rpt_type   => "Custom",
+            :db         => "VimPerformanceTrend",
+            :order      => "Ascending",
+            :sortby     => ["resource_name"],
+            :time_profile_id => @time_profile_exl.id,
+            :db_options => {:limit_col    => "max_derived_memory_available",
+                            :trend_col    => "derived_memory_used",
+                            :rpt_type     => "trend",
+                            :start_offset => 604800,
+                            :end_offset   => 0,
+                            :trend_db     => "HostPerformance",
+                            :interval     => "daily",
+                            :target_pcts  => [70, 80, 90]})
+        
+        t = Time.now
+        day_midnight = Time.new(t.year, t.month, t.day, 0, 0, 0, "+00:00").utc - 1.day
+        @used_mem_up = [700, 600, 500, 400]
+        @used_mem_up.size.times { |i|
+          @host1.metric_rollups << FactoryGirl.create(:metric_rollup_host_daily,
+                                                      :timestamp                => day_midnight - i.day,
+                                                      :time_profile_id          => @time_profile_all.id,
+                                                      :derived_memory_used      => @used_mem_up[i],
+                                                      :derived_memory_available => 1400,
+                                                      :capture_interval_name    => "daily")
```

Comment:
If this is the default in the factory, and it's not being verified in an expectation, is this needed here (and elsewhere)?